pkg/sync/manifest.go

```diff
@@ -1,55 +1,54 @@
 package sync
 
 import (
-	"encoding/json"
 	"fmt"
-	"strings"
 
-	"github.com/containers/image/manifest"
-	"github.com/opencontainers/go-digest"
+	"github.com/containers/image/v5/manifest"
 )
 
-// ManifestSchemaV2List describes a schema V2 manifest list
-type ManifestSchemaV2List struct {
-	Manifests []ManifestSchemaV2Info `json:"manifests"`
-}
-
-// ManifestSchemaV2Info includes of the imformation needes of a schema V2 manifest file
-type ManifestSchemaV2Info struct {
-	Digest string `json:"digest"`
-}
-
 // ManifestHandler expends the ability of handling manifest list in schema2, but it's not finished yet
 // return the digest array of manifests in the manifest list if exist.
-func ManifestHandler(m []byte, t string) (manifest.Manifest, []*digest.Digest, error) {
+func ManifestHandler(m []byte, t string, i *ImageSource) ([]manifest.Manifest, error) {
+
+	var manifestInfoSlice []manifest.Manifest
+
 	if t == manifest.DockerV2Schema2MediaType {
 		manifestInfo, err := manifest.Schema2FromManifest(m)
 		if err != nil {
-			return nil, nil, err
+			return nil, err
 		}
-		return manifestInfo, nil, nil
-	} else if strings.Contains(t, "application/vnd.docker.distribution.manifest.v1") {
+		manifestInfoSlice = append(manifestInfoSlice, manifestInfo)
+		return manifestInfoSlice, nil
+	} else if t == manifest.DockerV2Schema1MediaType || t == manifest.DockerV2Schema1SignedMediaType {
 		manifestInfo, err := manifest.Schema1FromManifest(m)
 		if err != nil {
-			return nil, nil, err
+			return nil, err
 		}
-		return manifestInfo, nil, nil
+		manifestInfoSlice = append(manifestInfoSlice, manifestInfo)
+		return manifestInfoSlice, nil
 	} else if t == manifest.DockerV2ListMediaType {
-		// get the digest of manifests in the manifest list
-		ml := ManifestSchemaV2List{}
-		if err := json.Unmarshal(m, &ml); err != nil {
-			return nil, nil, err
+		manifestSchemaListInfo, err := manifest.Schema2ListFromManifest(m)
+		if err != nil {
+			return nil, err
 		}
 
-		manifestDigests := []*digest.Digest{}
-		for _, item := range ml.Manifests {
-			d, err := digest.Parse(item.Digest)
-			if err != nil {
-				return nil, nil, err
+		for _, manifestDescriptorElem := range manifestSchemaListInfo.Manifests {
+
+			// 暂时至支持arm64/amd64架构的镜像同步
```

Comment:
Could you please use English comments here? That's more friendly for all of people. :)